test/lib.sh

```diff
@@ -59,3 +59,59 @@ EOF
 hr() {
     echo "========================="
 }
+
+diff() {
+	for file in `ls` ; do 
+		if [ -d $file ] ; then 
+			pushd $file  > /dev/null
+			diff $1/$file 
```

Comment:
Did you mean to check `$?` here?

(On a broader note, the more I use bash's `-e`, the less I like it. I would rather we were instead explicit about all our error conditions.)

Reply:
Removed `-e` from test case 01 (and avoided it in the new 02) via #14.